CRM/Contribute/BAO/Contribution.php

```diff
@@ -3218,7 +3219,7 @@ public static function recordFinancialAccounts(&$params, $financialTrxnValues =
       $params['trxnParams'] = $trxnParams;
 
       if (!empty($params['prevContribution'])) {
-        $updated = FALSE;
+        $updated = $ignoreChangeAmount = FALSE;
```

Comment:
I'm concerned that you are not setting $ignoreChangeAmount in all use cases. This PR also uses a function to set it below that is not yet merged in core. It would be good to indicate the dependence of this PR on that other one, perhaps in comment if not title.